File: API/UniUnboxdAPI/Models/DataTransferObjects/CourseCreationModel.cs
namespace UniUnboxdAPI.Models.DataTransferObjects;

// CourseCreationModel represents the data transfer object for creating a new course.
public class CourseCreationModel
{
    // Name of the course. Required field.
    public required string Name { get; set; }

    // Code of the course. Required field.
    public required string Code { get; set; }

    // Description of the course. Required field.
    public required string Description { get; set; }

    // Professor of the course. Required field.
    public required string Professor { get; set; }

    // Image URL of the course. Nullable field.
    public string? Image { get; set; }

    // Banner URL of the course. Nullable field.
    public string? Banner { get; set; } 

    // University ID of the course. Required field.
    public required int UniversityId { get; set; }
    
    // List of assigned professors for the course. Required field.
    public required List<AssignedProfessorModel> AssignedProfessors { get; set; }
}

File: Application/app/src/main/java/com/example/uniunboxd/models/CourseCreationModel.java
package com.example.uniunboxd.models;

public class CourseCreationModel {
    public final String Name;
    public final String Code;
    public final String Description;

    public final String Professor;

    public final String Image;
    public final String Banner;
    public final int UniversityID;

    public CourseCreationModel(String name, String code, String description, String professor, String image, String banner, int universityID) {
        Name = name;
        Code = code;
        Description = description;
        Professor = professor;
        Image = image;
        Banner = banner;
        UniversityID = universityID;
    }
}
